ogr/ogrsf_frmts/adbc/ogradbcbigquerylayer.cpp

```diff
@@ -0,0 +1,981 @@
+            osVal = nVal ? "TRUE" : "FALSE";
+        else
+            osVal = std::to_string(nVal);
+    }
+    else if (poFldDefn->GetType() == OFTReal)
+    {
+        osVal = CPLSPrintf("%.17g", poFeature->GetFieldAsDouble(iField));
+    }
+    else if (poFldDefn->GetType() == OFTDate)
+    {
+        char szTmpFieldValue[OGR_SIZEOF_ISO8601_DATETIME_BUFFER];
+        constexpr bool bAlwaysMillisecond = false;
+        OGRGetISO8601DateTime(poFeature->GetRawFieldRef(iField),
+                              bAlwaysMillisecond, szTmpFieldValue);
+        szTmpFieldValue[strlen("YYYY-MM-DD")] = 0;
+        osVal += "DATE \'";
+        osVal += szTmpFieldValue;
+        osVal += '\'';
+    }
+    else if (poFldDefn->GetType() == OFTDateTime)
+    {
+        osVal += '\'';
+        osVal += poFeature->GetFieldAsISO8601DateTime(iField, nullptr);
+        osVal += '\'';
+    }
+    else if (poFldDefn->GetType() == OFTBinary)
+    {
+        osVal += "b'";
+        int nCount = 0;
+        GByte *pabyVal = poFeature->GetFieldAsBinary(iField, &nCount);
+        osVal.reserve(nCount * 4 + 4);
+        for (int i = 0; i < nCount; ++i)
+        {
+            osVal += CPLSPrintf("\\x%02X", pabyVal[i]);
+        }
+        osVal += '\'';
+    }
+    else if (poFldDefn->GetType() == OFTStringList)
+    {
+        CSLConstList papszStr = poFeature->GetFieldAsStringList(iField);
+        osVal += '[';
+        for (int i = 0; papszStr && papszStr[i]; ++i)
+        {
+            if (i > 0)
+                osVal += ',';
+            osVal += '\'';
+            osVal += OGRDuplicateCharacter(papszStr[i], '\'');
+            osVal += '\'';
+        }
+        osVal += ']';
+    }
+    else if (poFldDefn->GetType() == OFTIntegerList)
+    {
+        int nCount = 0;
+        const int *panVals = poFeature->GetFieldAsIntegerList(iField, &nCount);
+        osVal += '[';
+        for (int i = 0; i < nCount; ++i)
+        {
+            if (i > 0)
+                osVal += ',';
+            osVal += std::to_string(panVals[i]);
+        }
+        osVal += ']';
+    }
+    else if (poFldDefn->GetType() == OFTInteger64List)
+    {
+        int nCount = 0;
+        const auto *panVals =
+            poFeature->GetFieldAsInteger64List(iField, &nCount);
+        osVal += '[';
+        for (int i = 0; i < nCount; ++i)
+        {
+            if (i > 0)
+                osVal += ',';
+            osVal += std::to_string(panVals[i]);
+        }
+        osVal += ']';
+    }
+    else if (poFldDefn->GetType() == OFTRealList)
+    {
+        int nCount = 0;
+        const double *padfVals =
+            poFeature->GetFieldAsDoubleList(iField, &nCount);
+        osVal += '[';
+        for (int i = 0; i < nCount; ++i)
+        {
+            if (i > 0)
+                osVal += ',';
+            osVal += CPLSPrintf("%.17g", padfVals[i]);
+        }
+        osVal += ']';
+    }
+    else
+    {
+        if (poFldDefn->GetSubType() == OFSTJSON)
+            osVal += "JSON ";
```

Comment:
Why is this necessary? Wouldn't be better to return the JSON as it is without prefixing and quoting?

Reply:
this is the way to insert a JSON litteral in a JSON column. Cf https://cloud.google.com/bigquery/docs/json-data?hl=en#create_a_json_value . Code comment added